backend/src/shared/infrastructure/core/application/command/RetryCommandBus.ts

```diff
@@ -0,0 +1,21 @@
+import { CommandBus } from '../../../../core/application/command/CommandBus';
+import { Command } from '../../../../core/application/command/Command';
+import { CommandResult } from '../../../../core/application/command/CommandResult';
+import { HasConstructor } from '../../../../../common/HasConstructor';
+import { CommandHandler } from '../../../../core/application/command/CommandHandler';
+
+export class RetryCommandBus implements CommandBus {
+  constructor(private readonly next: CommandBus, private readonly maxRetries: number) {}
+
+  async execute<CommandType extends Command>(command: CommandType): Promise<CommandResult> {
+    let result: CommandResult | undefined = undefined;
+    for (let retry = 0; (result === undefined || !result.isSuccess()) && retry < this.maxRetries; retry++) {
```

Comment:
A, widze, ze tutaj zmienilas warunek. Jak dla mnie to powinno zostac, albo jakos inaczej tylko test był zły.
Skoro to są maxRetries -to bym się spodziewał, że nawet jak tutaj damy 0, to raz się wykona. Ooo, taki test byłby fajny :) Bo teraz to jest maximum wywolan ze wszystkimi. Czyli w obecnym tescie expect powinien byc na 4 a nie 3.

Reply:
@lamparina - dobra, wrzucilem na to commita :)